core/lib/src/response/named_file.rs

```diff
@@ -38,25 +45,35 @@ impl NamedFile {
         // all of those `seek`s to determine the file size. But, what happens if
         // the file gets changed between now and then?
         let file = File::open(path.as_ref()).await?;
-        Ok(NamedFile(path.as_ref().to_path_buf(), file))
+        Ok(NamedFile {
+            path: path.as_ref().to_path_buf(),
+            file,
+            modified: None,
+        })
+    }
+
+    pub async fn with_last_modified_date<P: AsRef<Path>>(path: P) -> io::Result<NamedFile> {
+        let mut named_file = NamedFile::open(path).await?;
+        named_file.modified = Some(named_file.path().metadata()?.modified()?);
```

Comment:
This can be improved two ways:

```rust
named_file.modified = named_file.metadata().await?.modified().ok()
```

* get the metadata of the actually opened file, asynchronously
* fall back to `None` instead of an error on platforms where `modified()` isn't supported

Reply:
Done